backend/tomato/elements/external_network.py

```diff
@@ -192,6 +193,32 @@ def action_stop(self):
 			self.element = None
 		self.setState(ST_CREATED, True)
 
+	def checkMigrate(self):
+		#We only migrate if the external network is already started, otherwise we don't have to migrate
+		return self.state in [ST_PREPARED]
+	
+	def action_migrate(self,host):
+		from ..host import Host
+		
+		host = Host.objects.filter(name = host)
+		
+		UserError.check(host, code=UserError.NO_RESOURCES, message="Host not found",data={"type": self.TYPE})
+		UserError.check(self.element.host.name != host.name, code=UserError.UNSUPPORTED_ACTION, message="Migrating to the same host not allowed",data={"type": self.TYPE})
+		UserError.check(self.element.checkMigrate(), code=UserError.UNSUPPORTED_ACTION, message="Element can't be migrated",data={"type": self.TYPE})
+
+		kind = self.getParent().network.kind if self.parent and self.getParent().samenet else self.kind
```

Comment:
Below is all duplicated code, please create a method for this like ``_allocate(self, host)``